lib/routes/routeBackbeat.js

```diff
@@ -754,22 +794,94 @@ function deleteObjectTagging(request, response, log, callback) {
         dataStoreVersionId, log, callback);
 }
 
+function _createAzureConditionalDeleteObjectGetInfo(request) {
+    const { objectKey, bucketName, headers } = request;
+    const objectGetInfo = dataClient.toObjectGetInfo(
+        objectKey, bucketName, headers['x-scal-storage-class']);
+    return Object.assign({}, objectGetInfo, {
+        options: {
+            accessConditions: {
+                DateUnModifiedSince: new Date(headers['if-unmodified-since']),
+            },
+        },
+    });
+}
+
+function _azureConditionalDelete(request, response, log, cb) {
+    const objectGetInfo = _createAzureConditionalDeleteObjectGetInfo(request);
+    const reqUids = log.getSerializedUids();
+    return dataClient.delete(objectGetInfo, reqUids, err => {
+        if (err) {
+            log.error('error deleting object in Azure', {
+                error: err,
+                method: '_azureConditionalDelete',
+            });
+            return cb(err);
+        }
+        return _respond(response, {}, log, cb);
+    });
+}
+
+function _conditionalTagging(request, response, locations, log, cb) {
+    return _getLastModified(locations, log, (err, lastModified) => {
+        if (err) {
+            return cb(err);
+        }
+        const ifUnmodifiedSince = request.headers['if-unmodified-since'];
+        if (ifUnmodifiedSince !== lastModified) {
+            log.info('object has been modified, skipping tagging operation', {
+                method: '_conditionalTagging',
+                ifUnmodifiedSince,
+                lastModified,
+            });
+            return cb();
+        }
+        return handleTaggingOperation(
```

Comment:
When we reach this point, are we sure we're going to tag only nonversioned buckets? I don't see where we take this decision explicitly (I would say, relying on the fact that `if-unmodified-since` header is provided is not a proper check for this, I would rely on a version ID being passed in the location instead.

Reply:
It's conceived that the version check happens in backbeat, so this route would only be called with the `if-unmodified-since` header in the case that there is no version id from the source.

Reply:
Updated to decouple logic from the two services, as per the suggestion.